core/variant_setget.cpp

```diff
@@ -33,1994 +33,1377 @@
+				oob = true;                                                                                                          \
+				return;                                                                                                              \
+			}                                                                                                                        \
+			*VariantGetInternalPtr<m_elem_type>::get_ptr(value) = (*VariantGetInternalPtr<m_base_type>::get_ptr(base))[index];       \
+			oob = false;                                                                                                             \
+		}                                                                                                                            \
+		static void ptr_get(const void *base, int64_t index, void *member) {                                                         \
+			/* avoid ptrconvert for performance*/                                                                                    \
+			const m_base_type &v = *reinterpret_cast<const m_base_type *>(base);                                                     \
+			OOB_TEST(index, v.size());                                                                                               \
+			PtrToArg<m_elem_type>::encode(v[index], member);                                                                         \
+		}                                                                                                                            \
+		static void set(Variant *base, int64_t index, const Variant *value, bool &valid, bool &oob) {                                \
+			if (index < 0 || index >= VariantGetInternalPtr<m_base_type>::get_ptr(base)->size()) {                                   \
+				oob = true;                                                                                                          \
+				valid = false;                                                                                                       \
+				return;                                                                                                              \
+			}                                                                                                                        \
+			m_assign_type num;                                                                                                       \
+			if (value->get_type() == Variant::INT) {                                                                                 \
+				num = (m_assign_type)*VariantGetInternalPtr<int64_t>::get_ptr(value);                                                \
+			} else if (value->get_type() == Variant::FLOAT) {                                                                        \
+				num = (m_assign_type)*VariantGetInternalPtr<float>::get_ptr(value);                                                  \
```

Comment:
```suggestion
				num = (m_assign_type)*VariantGetInternalPtr<double>::get_ptr(value);                                                  \
```

I think this is right, since Variant floats are always 64-bit doubles, and you have `int64_t` above.